packages/flutter/lib/src/widgets/_web_image_web.dart

```diff
@@ -0,0 +1,344 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:collection';
+import 'dart:js_interop';
+import 'dart:ui_web' as ui_web;
+
+import '../../rendering.dart';
+import '../../widgets.dart';
+import '../foundation/synchronous_future.dart';
+import '../painting/_network_image_web.dart';
+import '../web.dart' as web;
+
+/// Returns [true] if the bytes at [url] can be fetched.
+///
+/// The bytes may be unable to be fetched if they aren't from the same origin
+/// and the server hosting them does not allow cross-origin requests.
+Future<bool> checkIfImageBytesCanBeFetched(String url) {
+  // First check if url is same origin.
+  final Uri uri = Uri.parse(url);
+  if (uri.origin == web.window.origin) {
+    return SynchronousFuture<bool>(true);
+  }
+
+  final Completer<web.XMLHttpRequest> completer =
+      Completer<web.XMLHttpRequest>();
+  final web.XMLHttpRequest request = httpRequestFactory();
+
+  request.open('HEAD', url, true);
+
+  request.addEventListener(
+      'load',
+      (web.Event e) {
+        final int status = request.status;
+        final bool accepted = status >= 200 && status < 300;
+        final bool fileUri = status == 0; // file:// URIs have status of 0.
+        final bool notModified = status == 304;
+        final bool unknownRedirect = status > 307 && status < 400;
+        final bool success =
+            accepted || fileUri || notModified || unknownRedirect;
+
+        if (success) {
+          completer.complete(request);
+        } else {
+          completer.completeError(e);
+        }
+      }.toJS);
+
+  request.addEventListener(
+      'error', ((JSObject e) => completer.completeError(e)).toJS);
+
+  request.send();
+
+  return completer.future.then((_) {
+    return true;
+  }, onError: (_) {
+    return false;
+  });
+}
+
+/// Returns a widget which displays the [src] in an <img> tag.
+Widget createImgElementWidget(
+  ImgElementProvider provider, {
+  Key? key,
+  ImageLoadingBuilder? loadingBuilder,
+  ImageFrameBuilder? frameBuilder,
+  ImageErrorWidgetBuilder? errorBuilder,
+}) {
+  return _ImgElementImage(
+    provider as _ImgElementProviderImpl,
+    key: key,
+    loadingBuilder: loadingBuilder,
+    frameBuilder: frameBuilder,
+    errorBuilder: errorBuilder,
+  );
+}
+
+ImgElementProvider createImgElementProvider(String src) =>
```

Comment:
@harryterkelsen It would be very useful if "String filter" added as parameter, in some cases additional transformations are needed for <img> to match it with "normal" NetworkImage

Reply:
I think instead of applying the CSS styling here, we should support filters in the engine by using normal Flutter rendering primitives. We plan to add more complete filter support for platform views in the Flutter Web engine soon.